### PR TITLE
Fix for #17, also catch ValueError when there is no password

### PR DESCRIPTION
**Pull request by treytabner**
These changes were required for Juno.

### CODE REVIEW

File: hybrid_assignment.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def list_projects_for_user(self, user_id, group_ids, hints):
         # Make sure the default project is in the project list for the user
         # user_id
         for project in projects:
-            if project.id == self.default_project_id:
+            if project.get('id') == self.default_project_id:
```
Comment (rhafer) [inline hybrid_assignment.py:101]: This was just fixed yesterday. See: https://github.com/SUSE-Cloud/keystone-hybrid-backend/pull/22

File: hybrid_identity.py [hunk 2]
```diff
@@ -60,7 +60,8 @@ def authenticate(self, user_id, password):
             assert utils.check_password(password, user_ref['password'])
         except TypeError:
             raise AssertionError('Invalid user / password')
-        except KeyError:  # if it doesn't have a password, it must be LDAP
+        except (ValueError, KeyError):
+            # if it doesn't have a password, it must be LDAP
```
Comment (rhafer) [inline hybrid_identity.py:64]: Hm, under which conditions does this throw a `ValueError`? Isn't `KeyError` already enough to catch the case where user_ref has no password?

**Comment by rhafer** [comment]
@treytabner thanks a lot for you contribution. See my comments inline. Please note that I'll like merge https://github.com/SUSE-Cloud/keystone-hybrid-backend/pull/21  (or a variation of that) soon. So the only remaining issue will be the `KeyError` thing. If that really is an issue.

File: hybrid_identity.py (hunk 2 above)
Comment (treytabner) [inline hybrid_identity.py:64]: I was getting a ValueError, I'll do some more testing, thank you.

Comment (treytabner) [inline hybrid_identity.py:64]: The error I get without catching ValueError is. If you have any insight on the error, I would appreciate any information.

2015-01-16 18:06:57.958 20417 INFO passlib.registry [-] registered crypt handler 'sha512_crypt': <class 'passlib.handlers.sha2_crypt.sha512_crypt'>
2015-01-16 18:06:58.011 20417 ERROR keystone.common.wsgi [-] invalid sha512-crypt hash
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi Traceback (most recent call last):
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/common/wsgi.py", line 223, in **call**
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     result = method(context, *_params)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/token/controllers.py", line 100, in authenticate
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     context, auth)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/token/controllers.py", line 296, in _authenticate_local
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     password=password)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/notifications.py", line 307, in wrapper
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     result = f(wrapped_self, context, user_id, *args, *_kwargs)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/identity/core.py", line 202, in wrapper
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     return f(self, _args, *_kwargs)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/identity/core.py", line 213, in wrapper
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     return f(self, _args, *_kwargs)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/identity/core.py", line 549, in authenticate
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     ref = driver.authenticate(entity_id, password)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/identity/backends/hybrid_identity.py", line 63, in authenticate
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     assert utils.check_password(password, user_ref['password'])
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/common/utils.py", line 139, in check_password
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     return passlib.hash.sha512_crypt.verify(password_utf8, hashed)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/passlib/utils/handlers.py", line 464, in verify
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     self = cls.from_string(hash)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/passlib/handlers/sha2_crypt.py", line 454, in from_string
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     raise ValueError("invalid sha512-crypt hash")
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi ValueError: invalid sha512-crypt hash
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi 
2015-01-16 18:06:58.038 20417 INFO eventlet.wsgi.server [-] 10.90.1.10 - - [16/Jan/2015 18:06:58] "POST /v2.0/tokens HTTP/1.1" 500 398 2.412308

**Pull request closed without merging**